pkg/vpc/vpc.go

```diff
@@ -300,7 +301,42 @@ func UseFromClusterStack(ctx context.Context, provider api.ClusterProvider, stac
 		return strings.Split(v, ",")
 	}
 	importSubnetsFromIDList := func(subnetMapping api.AZSubnetMapping, value string) error {
-		return ImportSubnetsFromIDList(ctx, provider.EC2(), spec, subnetMapping, splitOutputValue(value))
+		var (
+			vpcSubnets   []string
+			stackSubnets []string
+			toBeImported []string
+		)
+		// collect VPC subnets as returned by CFN stack outputs
+		stackSubnets = splitOutputValue(value)
+
+		// collect VPC subnets as returned by EC2 API
+		ec2API := provider.EC2()
+		output, err := ec2API.DescribeSubnets(ctx, &ec2.DescribeSubnetsInput{
+			Filters: []ec2types.Filter{
+				{
+					Name:   aws.String("vpc-id"),
+					Values: []string{spec.VPC.ID},
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+		for _, o := range output.Subnets {
+			vpcSubnets = append(vpcSubnets, *o.SubnetId)
+		}
+
+		// if a subnet is present on the stack outputs, but actually missing from VPC
+		// e.g. it was manually deleted by the user using AWS CLI/Console
+		// than log a warning and don't import it into cluster spec
+		for _, ssID := range stackSubnets {
+			if !slices.Contains(vpcSubnets, ssID) {
+				logger.Warning("%s was found on cluster cloudformation stack, but has been removed from %s outside of eksctl", ssID, spec.VPC.ID)
```

Comment:
Minor nit:

```suggestion
				logger.Warning("%s was found on cluster's CloudFormation stack outputs, but has been removed from VPC %s outside of eksctl", ssID, spec.VPC.ID)
```

Reply:
Also, when this condition holds true, we should also log the set of subnets that will be used for the nodegroup.